File: MML_LANGUAGE/mml-webapp/upload/mml.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import pandas as pd 
import numpy as np
import json
import warnings
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn import tree
warnings.filterwarnings("ignore")
mml_data = pd.read_csv("upload/Spotify.csv", sep=',', engine='python')
y = mml_data.loc[:,[like']]
X =  mml_data.loc[:,['NumId','danceability','energy','mode','tempo','time_signature']]
results = []
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.19999999, random_state=0)
results.append({})
results[0]["output"] = []
results[0]["model"] = "Decision tree"
clf = tree.DecisionTreeClassifier(criterion = "gini", max_depth = 1)
clf.fit(X_train, y_train)
y_pred=clf.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
results[0]["output"].append({"metric" : "accuracy", "value" : accuracy})
print(json.dumps(results))

File: MML_LANGUAGE/mml-webapp/upload/mml.R
library(utils)
library(dplyr, warn.conflicts = FALSE)
library(caret)
library(e1071)
library(rpart)
library(rjson)
data <- read.csv("upload/iris.csv", sep=',')
y <- "variety"
X <- colnames(data %>% select(-"variety"))
formula = as.formula(paste(y, paste(X,collapse = " + "), sep = " ~ "))
results = list()
set <- sample(1:nrow(data),floor(0.8*nrow(data)))
dataTrain <- data[set,]
dataTest <- data[-set,]
results[[1]] = list(model=c(), output=list())
results[[1]]$model = "Decision tree"
clf = rpart(formula = formula,data = dataTrain, method = "class",control = (maxdepth = 1),parms = list(split ="gini"))
y_pred = predict(clf, dataTest, type = "class")
mat_conf <- table(y_pred,unlist(dataTest %>% select(y)))
accuracy = sum(diag(mat_conf))/sum(mat_conf)
results[[1]]$output[[1]] = list(metric = "accuracy", value = accuracy)
print(toJSON(results))
